connectors/sources/sharepoint_server.py

```diff
@@ -375,28 +358,25 @@ async def _fetch_data_with_next_url(
         next_url = ""
         while True:
             if next_url != "":
-                response = await anext(
-                    self._api_call(
-                        url_name=param_name,
-                        url=next_url,
-                        selected_field=selected_field,
-                    )
-                )
+                async for response in self._api_call(
```

Comment:
Why `async for`? Is result of `_api_call` a collection?

Reply:
The expected for _api_call is collection only, but if I use anext then I have to handle `StopAsyncIteration` exception explicitly, that's why I use async for here.

Reply:
I see, ideally you should refactor _api_call to two methods: one for a single entity, one for the collection. This is outside of the scope of this PR, however